src/stripeLinter.ts

```diff
@@ -73,33 +75,30 @@ export const shouldSearchFile = (currentFilename: string): boolean => {
 
 export class StripeLinter {
   telemetry: Telemetry;
+  git: Git;
 
-  constructor(telemetry: Telemetry) {
+  constructor(telemetry: Telemetry, git: Git) {
     this.telemetry = telemetry;
+    this.git = git;
   }
 
-  activate() {
-    this.lookForHardCodedAPIKeys();
+  async activate() {
+    if (window.activeTextEditor) {
+      await this.lookForHardCodedAPIKeys(window.activeTextEditor.document);
+    }
     workspace.onDidSaveTextDocument(this.lookForHardCodedAPIKeys);
   }
 
-  lookForHardCodedAPIKeys = (): void => {
-    const editor = window.activeTextEditor;
-    if (!editor) { return; }
-
-    const {document} = editor;
-
+  lookForHardCodedAPIKeys = async (document: TextDocument): Promise<void> => {
     const currentFilename = document.uri.path;
     if (!shouldSearchFile(currentFilename)) { return; }
 
     const text = document.getText();
     const lines = text.split('\n');
 
-    // get each line's possible API key warnings
-    // then flatten nested diagnostics into a flat array
-    const fileDiagnostics: Diagnostic[] = lines
-      .map(this.prepareLineDiagnostics)
-      .reduce((acc, next) => acc.concat([...next]), []);
+    const message = await this.git.isGitRepo(document.uri) ? diagnosticMessageGit : diagnosticMessageNoGit;
```

Comment:
Would it be reasonable to assume that this won't change over the lifecycle of the extension running?
i.e. Could we save some time by calculating this outside of StripeLinter and passing in just an `isGitRepo` flag to StripeLinter's constructor?

Reply:
Good idea, I don't expect a user to be initializing git or removing git in the middle of a session very often, so we can optimize for the common case and only compute `isGitRepo` once when the extension activates.

I also plan to add more functionality to the `Git` class such as checking if a file is git-ignored (the main thing to do to remove our dependency on the git vscode extension), so I want to continue passing the `Git` instance to `StripeLinter` if that's reasonable.

Reply:
Actually this could get hairy when using [multi-root workspaces](https://code.visualstudio.com/docs/editor/multi-root-workspaces). We'd have to do this per folder in the workspace. I did some benchmarking and found that `access`ing the `.git` every time adds only a few ms of latency, so I think it's okay. What are your thoughts?

Reply:
Ah good point. Yeah that seems like the best option.